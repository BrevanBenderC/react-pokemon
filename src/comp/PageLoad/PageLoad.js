import React from 'react';
import RenderPokes from '../renderPokes/renderPokes';
import { Button } from '@mui/material';

export default function PokeList({ pokemon, setCurrentPage, setLoading }) {
  const handleNextPage = () => {
    setCurrentPage((prevState) => ++prevState);
    setLoading(true);
  };

  const handlePrevPage = () => {
    setCurrentPage((prevState) => --prevState);
    setLoading(true);
  };

  return (
    <>
      <div className="compiledMon">
        {pokemon.map((pokemon) => (
          <RenderPokes key={pokemon.id} {...pokemon} />
        ))}
      </div>

      <div className="button">
        <Button variant="outlined" onClick={handlePrevPage}>
          Previous Page
        </Button>
        <Button variant="outlined" onClick={handleNextPage}>
          Next Page
        </Button>
      </div>
    </>
  );
}
