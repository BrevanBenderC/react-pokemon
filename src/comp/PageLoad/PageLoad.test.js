import { render } from '@testing-library/react';
import PageLoad from './PageLoad';

test('should render the list of poke cards, be able to update page and loading state', () => {
  const container = render(
    <PageLoad pokemon={['butterfree', 'pikachu']} setLoading="false" setCurrentPage="1" />
  );
  expect(container).toMatchSnapshot();
});
