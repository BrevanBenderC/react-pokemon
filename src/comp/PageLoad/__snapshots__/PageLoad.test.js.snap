// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the list of poke cards, be able to update page and loading state 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="compiledMon"
      >
        <div
          class="indiPoke"
        >
          <img
            class="pokeImg"
          />
          <h1
            class="name"
          />
          <span
            class="stats"
          >
            Hp: 
          </span>
          <span
            class="stats"
          >
            Attack: 
          </span>
          <span
            class="stats"
          >
            Defense: 
          </span>
          <span
            class="stats"
          >
            Sp Attack: 
          </span>
          <span
            class="stats"
          >
            Sp Defense: 
          </span>
          <span
            class="stats"
          >
            Speed: 
          </span>
        </div>
        <div
          class="indiPoke"
        >
          <img
            class="pokeImg"
          />
          <h1
            class="name"
          />
          <span
            class="stats"
          >
            Hp: 
          </span>
          <span
            class="stats"
          >
            Attack: 
          </span>
          <span
            class="stats"
          >
            Defense: 
          </span>
          <span
            class="stats"
          >
            Sp Attack: 
          </span>
          <span
            class="stats"
          >
            Sp Defense: 
          </span>
          <span
            class="stats"
          >
            Speed: 
          </span>
        </div>
      </div>
      <div
        class="button"
      >
        <button
          class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Previous Page
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Next Page
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="compiledMon"
    >
      <div
        class="indiPoke"
      >
        <img
          class="pokeImg"
        />
        <h1
          class="name"
        />
        <span
          class="stats"
        >
          Hp: 
        </span>
        <span
          class="stats"
        >
          Attack: 
        </span>
        <span
          class="stats"
        >
          Defense: 
        </span>
        <span
          class="stats"
        >
          Sp Attack: 
        </span>
        <span
          class="stats"
        >
          Sp Defense: 
        </span>
        <span
          class="stats"
        >
          Speed: 
        </span>
      </div>
      <div
        class="indiPoke"
      >
        <img
          class="pokeImg"
        />
        <h1
          class="name"
        />
        <span
          class="stats"
        >
          Hp: 
        </span>
        <span
          class="stats"
        >
          Attack: 
        </span>
        <span
          class="stats"
        >
          Defense: 
        </span>
        <span
          class="stats"
        >
          Sp Attack: 
        </span>
        <span
          class="stats"
        >
          Sp Defense: 
        </span>
        <span
          class="stats"
        >
          Speed: 
        </span>
      </div>
    </div>
    <div
      class="button"
    >
      <button
        class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Previous Page
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Next Page
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
